src/commands/tcp-redirs.js

```diff
@@ -29,11 +32,31 @@ async function list (params) {
   }
 }
 
+async function acceptPayment (result, skipConfirmation) {
+  if (!skipConfirmation) {
+    result.lines.forEach(({ description, VAT, price }) => Logger.println(`${description}\tVAT: ${VAT}%\tPrice: ${price}€`));
+    Logger.println(`Total (without taxes): ${result.totalHT}€`);
+    Logger.println(colors.bold(`Total (with taxes): ${result.totalTTC}€`));
+
+    await Interact.confirm(
+      `You're about to pay ${result.totalTTC}€, confirm? (yes or no) `,
+      'No confirmation, aborting TCP redirection creation',
+    );
+  }
+}
+
 async function add (params) {
-  const { alias, namespace } = params.options;
+  const { alias, namespace, yes: skipConfirmation } = params.options;
   const { ownerId, appId } = await AppConfig.getAppDetails({ alias });
 
-  const { port } = await application.addTcpRedir({ id: ownerId, appId }, { namespace }).then(sendToApi);
+  const { port } = await application.addTcpRedir({ id: ownerId, appId }, { namespace }).then(sendToApi).catch((error) => {
+    if (error.status === 402) {
+      return acceptPayment(error.response.body, skipConfirmation).then(() => application.addTcpRedir({ id: ownerId, appId, payment: 'accepted' }, { namespace })).then(sendToApi);
```

Comment:
This line is a bit long IMO.

What about:

```js
if (error.status === 402) {
  return acceptPayment(error.response.body, skipConfirmation).then(() => {
    application.addTcpRedir({ id: ownerId, appId, payment: 'accepted' }, { namespace })).then(sendToApi);
  }
}
```

I think a try/catch would be even more readable but that's just me. The then/catch is fine.

Reply:
Also, it helped me realize, in your code, the `sendToApi` is chained on the return of `acceptPayment` and note `application.addTcpRedir`. In this case, it changes nothing but it may introduce confusion.

Reply:
Fixed